tests/chart_tests/test_logging_sidecar.py

```diff
@@ -42,3 +44,50 @@ def test_logging_sidecar_config_disabled(self, kube_version):
             show_only="templates/logging-sidecar-configmap.yaml",
         )
         assert len(docs) == 0
+
+    def test_logging_sidecar_custom_config(self, kube_version):
+        """Test logging sidecar config with customConfig flag enabled"""
+        custom_side_carconfig = textwrap.dedent(
+            """
+        loggingSidecar:
+          enabled: true
+          customConfig: true
+          name: sidecar-logging-consumer
+          airflowSidecarConfig: |2
+              log_schema:
+                timestamp_key : "@timestamp"
+              data_dir: "${SIDECAR_LOGS}"
+              sources:
+                generate_syslog:
+                  type: file
+                  include:
+                    - "${SIDECAR_LOGS}/*.log"
+                  read_from: beginning
+              transforms:
+                transform_syslog:
+                  type: add_fields
+                  inputs:
+                    - generate_syslog
+                  fields:
+                    component: "${COMPONENT:--}"
+                    workspace: "${WORKSPACE:--}"
+                    release: "${RELEASE:--}"
+              sinks:
+                out:
+                  type: datadog
+                  inputs:
+                    - transform_syslog
+                """
+        )
```

Comment:
Another way we've done this that is more compact is to use json for this. https://github.com/astronomer/astronomer/blob/c66c8f75e46d5d6ec874c5631a92ab3890ed927d/tests/chart_tests/test_ingress.py#L31-L39

Reply:
Took inspiration from this - https://github.com/apache/airflow/blob/9e2658ea8850b0bacba8a3fe82c457dbeb48d591/tests/charts/test_extra_configmaps_secrets.py#L32
Agree the above is more compact